ts/features/fims/history/saga/handleGetFimsHistorySaga.ts

```diff
@@ -0,0 +1,40 @@
+import * as E from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/function";
+import { call, put } from "typed-redux-saga/macro";
+import { ActionType } from "typesafe-actions";
+import { SagaCallReturnType } from "../../../../types/utils";
+import { withRefreshApiCall } from "../../../fastLogin/saga/utils";
+import { FimsHistoryClient } from "../api/client";
+import { fimsHistoryGet } from "../store/actions";
+
+export function* handleGetFimsHistorySaga(
+  getFimsHistory: FimsHistoryClient["getConsents"],
+  bearerToken: string,
+  action: ActionType<typeof fimsHistoryGet.request>
+) {
+  const getHistoryRequest = getFimsHistory({
+    Bearer: bearerToken,
+    continuationToken: action.payload.continuationToken
+  });
+
+  try {
+    const getHistoryResult = (yield* call(
+      withRefreshApiCall,
+      getHistoryRequest,
+      action
+    )) as SagaCallReturnType<typeof getFimsHistory>;
+
+    yield pipe(
+      getHistoryResult,
+      E.fold(
+        error => put(fimsHistoryGet.failure(error.toString())),
```

Comment:
If done like this, later saga-testing will be difficult since we cannot use the testing api to check for the put action. 

What about creating a function that, given `getHistoryResult`, extracts the content we are interested in and returns an action to dispatch?

Such function can be called with a `yield * call` and its result can be executed with `yield * put(...)`, simplifying the saga testing